pkg/session_recording/recordings.jsx

```diff
@@ -505,15 +484,17 @@
         }
 
         handleDateSinceChange(date) {
-            this.setState({dateSince: date});
-            this.clearRecordings();
-            this.journalctlRestart();
+            this.setState({dateSince: date}, () => {
+                this.clearRecordings();
+                this.journalctlRestart();
+            });
         }
 
         handleDateUntilChange(date) {
-            this.setState({dateUntil: date});
-            this.clearRecordings();
-            this.journalctlRestart();
+            this.setState({dateUntil: date}, () => {
+                this.clearRecordings();
+                this.journalctlRestart();
+            });
```

Comment:
At this moment the React documentation says componentDidUpdate use is recommended over the callback use: https://facebook.github.io/react/docs/react-component.html#setstate

Regardless, it is better to avoid callbacks whenever possible, because with them it is harder to reason about the exact sequence of events and their relative order, and thus they are harder to debug.

Could you please try to put those into componentDidUpdate, and if you see any difficulties, ping me and we'll debug this together.